californium-core/src/main/java/org/eclipse/californium/core/CoapServer.java

```diff
@@ -305,7 +309,7 @@ public MessageDeliverer getMessageDeliverer() {
 	 * @param endpoint the endpoint to add
 	 */
 	@Override
-	public void addEndpoint(final Endpoint endpoint) {
+	public final void addEndpoint(final Endpoint endpoint) {
```

Comment:
This is same as point 1 so lets park it for now until we have more structured solution.